daemon/api_aliases_test.go

```diff
@@ -343,6 +344,23 @@ func (s *aliasesSuite) TestUnaliasAliasSuccess(c *check.C) {
 	st.Unlock()
 	c.Assert(err, check.IsNil)
 
+	// Wait for the change to become clean
+	testTimeout := time.After(time.Second * 2)
+Loop:
+	for {
+		select {
+		case <-testTimeout:
+			c.Fatalf("timeout waiting for change to be clean")
+		case <-time.After(1 * time.Millisecond):
+			st.Lock()
+			done := chg.IsClean()
+			st.Unlock()
+			if done {
+				break Loop
+			}
+		}
+	}
+
```

Comment:
Although it's only 2 instances, but it's a larger block of code, so maybe a helper, eg. `waitUntilClean(c *Check, st *state.State, chg *state.Change)` ?

Reply:
Thanks, I applied something very similar. I also rebased the branch because if had grown to a lot of (reverted) commits. :-)

I will still add one or maybe more commits to address the comment by Samuele about uniforming the same logic to other places.